src/System.Configuration.ConfigurationManager/tests/System/Configuration/StringValidatorTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Configuration;
+using Xunit;
+
+namespace System.ConfigurationTests
+{
+    public class StringValidatorTests
+    {
+        [Fact]
+        public void Constructor_MinLength()
+        {
+            //We don't expect this to fail
+            StringValidator validator = new StringValidator(5);
+        }
+
+        [Fact]
+        public void Constructor_MinValueAndMaxValue()
+        {
+            // This should complete with no errors.
+            StringValidator validator = new StringValidator(5, 10);
+        }
+
+        [Fact]
+        public void Constructor_MinValueMaxValueAndInvalidChars()
+        {
+            // this should complete with no errors.
+            StringValidator validator = new StringValidator(5, 10, "abcde");
+        }
+
+        [Fact]
+        public void CanValidate_PassInStringType()
+        {
+            StringValidator validator = new StringValidator(5);
+            bool result = validator.CanValidate(typeof(string));
+            Assert.True(result);
+        }
+
+        [Fact]
+        public void CanValidate_PassInNotStringType()
+        {
+            StringValidator validator = new StringValidator(5);
+            bool result = validator.CanValidate(typeof(Int32));
+            Assert.False(result);
+        }
+
+        [Fact]
+        public void Validate_PassInNonString()
+        {
+            StringValidator validator = new StringValidator(5);
+            ArgumentException thrownException = Assert.Throws<ArgumentException>(() => validator.Validate(5));
+            ArgumentException expected = new ArgumentException(SR.Validator_value_type_invalid, string.Empty);
+            Assert.Equal(expected.Message, thrownException.Message);
+        }
+
+        [Fact]
+        public void Validate_StringTooSmall()
+        {
+            StringValidator validator = new StringValidator(5);
+            ArgumentException thrownException = Assert.Throws<ArgumentException>(() => validator.Validate("Hi"));
+            ArgumentException expected = new ArgumentException(string.Format(SR.Validator_string_min_length, 5), string.Empty);
+            Assert.Equal(expected.Message, thrownException.Message);
+        }
+
+        [Fact]
+        public void Validate_StringTooBig()
+        {
+            StringValidator validator = new StringValidator(5, 10);
+            ArgumentException thrownException = Assert.Throws<ArgumentException>(() => validator.Validate("This is more than ten"));
+            ArgumentException expected = new ArgumentException(string.Format(SR.Validator_string_max_length, 10), string.Empty);
+            Assert.Equal(expected.Message, thrownException.Message);
```

Comment:
No need to compare the exception objects. We just care about the type and that it is actually been thrown on that code path, as the object comparison could check for the param name, exception message, stack trace, etc. All those thing could change over target frameworks, so Assert.Throws<ExceptionType>(action) is enough. 

In some cases we test for the param name as well as we want to give the correct param names to the user and that should be the same across target groups (we find places where desktop param names aren't the same as we didn't paid attention on them by that time but when we ported to .Net Core we added them correctly)